core/jobs/batch_jobs/store_profile_images_to_gcs_jobs.py

```diff
@@ -0,0 +1,203 @@
+        """
+        file_dict = {}
+        username = user_model.username
+        filename = f'user/{username}/profile_picture.png'
+        profile_picture_binary = utils.convert_png_data_url_to_binary(
+            user_model.profile_picture_data_url)
+        file_dict = {
+            'filepath': filename,
+            'data': profile_picture_binary
+        }
+        return file_dict
+
+    def _filenames_webp(
+        self, user_model: user_models.UserSettingsModel
+    ) -> gcs_io.FileObjectDict:
+        """Returns file object for webp images to write to the GCS.
+
+        Args:
+            user_model: UserSettingsModel. The user settings model.
+
+        Returns: file_dict. The FileObjectDict.
+        """
+        file_dict = {}
+        username = user_model.username
+        filename = f'user/{username}/profile_picture.webp'
+        profile_picture = user_model.profile_picture_data_url
+        webp_base64 = webptools.base64str2webp_base64str(
+            base64str=profile_picture, image_type="png",
+            option="-q 80",logging="-v")
+        webp_binary = utils.convert_webp_data_url_to_binary(webp_base64)
+        file_dict = {
+            'filepath': filename,
+            'data': webp_binary
+        }
+        return file_dict
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        users_with_valid_username = (
+            self.pipeline
+            | 'Get all non-deleted UserSettingsModel' >> ndb_io.GetModels(
+                user_models.UserSettingsModel.get_all(include_deleted=False))
+            | 'Filter valid users with not None username' >> beam.Filter(
+                lambda model: model.username is not None)
+        )
+
+        write_png_files_to_gcs = (
+            users_with_valid_username
+            | 'Map files for png' >> beam.Map(self._filenames_png)
+            | 'Write png file to GCS' >> gcs_io.WriteFile(
+                mode='wb', mime_type='image/png')
+        )
+
+        total_png_images = (
+            write_png_files_to_gcs
+            | 'Total png images wrote to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL PNG IMAGES'))
+        )
+
+        write_webp_files_to_gcs = (
+            users_with_valid_username
+            | 'Map files for webp' >> beam.Map(self._filenames_webp)
+            | 'Write webp file to GCS' >> gcs_io.WriteFile(
+                mode='wb', mime_type='image/webp')
+        )
+
+        total_webp_images = (
+            write_webp_files_to_gcs
+            | 'Total webp images wrote to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL WEBP IMAGES'))
+        )
+
+        return (
+            (
+                total_png_images,
+                total_webp_images
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+
+class AuditProfilePictureFromGCSJob(base_jobs.JobBase):
```

Comment:
I think this job should make sure that the saved file and the `profile_picture_data_url` are the same, right? (with some transformations ofc)

Reply:
Done